app/jsx/pages/SearchPage.jsx

```diff
@@ -360,72 +366,91 @@ class FacetFieldset extends React.Component {
     }
   }
 
```

Comment:
Logic in code that follows has been drastically changed in order to better display checked facets.